test_conformance/basic/test_imagedim.cpp

```diff
@@ -41,8 +41,13 @@ static const char *image_dim_kernel_code =
 static unsigned char *
 generate_8888_image(int w, int h, MTdata d)
 {
-    unsigned char   *ptr = (unsigned char*)malloc(w * h * 4);
-    int             i;
+    unsigned char *ptr =
+        (unsigned char *)malloc(sizeof(unsigned char) * 4 * w * h);
```

Comment:
I don't think that adding the multiplication by `sizeof(unsigned char)` will hurt anything, but is it required?  I believe `sizeof(unsigned char)` is required to be 1:

https://en.cppreference.com/w/c/language/sizeof

> sizeof(char), sizeof(signed char), and sizeof(unsigned char) always return 1.

Reply:
Hi Ben,
We seem to be exceeding limit 32-bit limit for many of our higher end GPUs where typical size of image tested is 32k x 32k. This change fixes the issue. This way lying around in our internal CTS version. So, upstreamed as is.
The change description is bit misleading.
I believe what fixes the issue is not the value of sizeof(unsigned char), but the fact that this change makes size computation 64-bit instead of 32-bit which caused the overflow.
A cleaner fix would be to use long int or size_t for inputs to this test, but that may be somewhat bigger change. Will check and update PR with a cleaner fix.

Reply:
I see, makes sense - the important part is the multiplication by `size_t` which changes the calculation from a 32-bit multiplication to a 64-bit multiplication.

https://godbolt.org/z/PPK7sY

Thanks!